Use intptr_t for pointer alignment arithmetic

ALIGN() cast the pointer to long, which is 32 bits on LLP64 targets
(64-bit Windows), so the upper half of the address was discarded before
taking the remainder. Use intptr_t.


diff --git a/lib/tre-internal.h b/lib/tre-internal.h
index 572623d..0c56344 100644
--- a/lib/tre-internal.h
+++ b/lib/tre-internal.h
@@ -18,6 +18,7 @@
 #endif /* !HAVE_WCTYPE_H */
 
 #include <limits.h>
+#include <stdint.h>
 #include <ctype.h>
 #include "../local_includes/tre.h"
 
@@ -132,8 +133,8 @@ typedef enum { STR_WIDE, STR_BYTE, STR_MBS, STR_USER } tre_str_type_t;
 /* Returns number of bytes to add to (char *)ptr to make it
    properly aligned for the type. */
 #define ALIGN(ptr, type) \
-  ((((long)ptr) % sizeof(type)) \
-   ? (sizeof(type) - (((long)ptr) % sizeof(type))) \
+  ((((intptr_t)ptr) % sizeof(type)) \
+   ? (sizeof(type) - (((intptr_t)ptr) % sizeof(type))) \
    : 0)
 
 #undef MAX
